libs/service-portal/settings/personal-information/src/screens/EditPhoneNumber/EditPhoneNumber.tsx

```diff
@@ -57,14 +66,27 @@ export const EditPhoneNumber: ServicePortalModuleComponent = () => {
       if (!parsePhoneNumber?.isValid()) {
         throw Error('Not valid phone number')
       }
-      await updateIslykill({
-        variables: {
-          input: {
-            mobile: `+${parsePhoneNumber?.countryCallingCode}-${parsePhoneNumber?.nationalNumber}`,
-            email: settings?.email,
+
+      if (settings) {
+        await updateIslykill({
+          variables: {
+            input: {
+              mobile: `+${parsePhoneNumber?.countryCallingCode}-${parsePhoneNumber?.nationalNumber}`,
+              email: settings?.email,
+            },
           },
-        },
-      })
+        })
+        await updateUserProfile({
+          mobilePhoneNumber: formData.tel,
+        })
+      } else {
+        await createIslykillSettings({
+          mobile: `+${parsePhoneNumber?.countryCallingCode}-${parsePhoneNumber?.nationalNumber}`,
+        })
+        await createUserProfile({
+          mobilePhoneNumber: formData.tel,
+        })
+      }
```

Comment:
I think it would be better to remove the Islykill domain and instead wrap it behind the user profile domain/service.

We also need to verify the phone number, right?

Reply:
We do want to verify the phone number yes. Phone verification is included here. The only way for the user to call this is to have already verified beforehand.